### PR TITLE
GEODE-9064: Configure serial filter during ManagementAgent start

### PR DESCRIPTION
**Pull request by kirklund**
The Geode JMX layer will configure the system property
“jmx.remote.rmi.server.serial.filter.pattern" to only accept
JDK classes identified as open-types required for JMX.
    
If the system property already has a value then Geode will log
a statement and leave the system property alone.

If the Java version is Java 8 then the system property does
not exist and will not be set.

Further pull requests may be submitted for Java 8.

### CODE REVIEW

**Review by onichols-pivotal — APPROVED** [review]

**Review by rhoughton-pivot — APPROVED** [review]

**Review by jinmeiliao — CHANGES_REQUESTED** [review]

File: geode-assembly/src/acceptanceTest/java/org/apache/geode/management/LocatorManagerConfiguresSerialFilterAcceptanceTest.java [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import org.apache.geode.management.internal.OpenJmxTypesSerialFilter;
+import org.apache.geode.test.assertj.LogFileAssert;
+import org.apache.geode.test.junit.rules.RequiresGeodeHome;
+
+public class LocatorManagerConfiguresSerialFilterAcceptanceTest {
```
[the remark this answers is not in the document]
Comment (kirklund) [reply geode-assembly/src/acceptanceTest/java/org/apache/geode/management/LocatorManagerConfiguresSerialFilterAcceptanceTest.java]: Done

File: geode-assembly/src/acceptanceTest/java/org/apache/geode/management/ServerManagerConfiguresSerialFilterAcceptanceTest.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import org.apache.geode.management.internal.OpenJmxTypesSerialFilter;
+import org.apache.geode.test.assertj.LogFileAssert;
+import org.apache.geode.test.junit.rules.RequiresGeodeHome;
+
+public class ServerManagerConfiguresSerialFilterAcceptanceTest {
```
[the remark this answers is not in the document]
Comment (kirklund) [reply geode-assembly/src/acceptanceTest/java/org/apache/geode/management/ServerManagerConfiguresSerialFilterAcceptanceTest.java]: Done

File: geode-core/src/main/java/org/apache/geode/management/internal/JmxRmiSerialFilter.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.management.internal;
+
+@FunctionalInterface
+public interface JmxRmiSerialFilter {
+
+  /**
+   * Configure the “jmx.remote.rmi.server.serial.filter.pattern” system property if Java version is
```
[the remark this answers is not in the document]
Comment (kirklund) [reply geode-core/src/main/java/org/apache/geode/management/internal/JmxRmiSerialFilter.java]: I removed javadocs from JmxRmiSerialFilter. If I think of some better javadocs, I'll add them in.

File: geode-core/src/main/java/org/apache/geode/management/internal/JmxRmiSerialFilter.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+   * greater than Java 8. If the system property already has a non-null value with a length greater
+   * than zero, then leave it as is.
+   */
+  void configureSerialFilterIfEmpty();
```
[the remark this answers is not in the document]
Comment (kirklund) [reply geode-core/src/main/java/org/apache/geode/management/internal/JmxRmiSerialFilter.java]: I renamed this `configureSerialFilter`. I also renamed some classes and test classes to hopefully have better names.

**Review by jinmeiliao — APPROVED** [review]

**Pull request merged**